### PR TITLE
Cortx 33258 part 1

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
2 tests, one for diff pod every time and one for same pod in loop

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 1]
```diff
@@ -1372,195 +1372,230 @@ def test_pod_restart_node_nw_down(self):
...
-
-            LOGGER.info("Step 5: Create s3 account, create multiple buckets and run IOs")
-            users = self.mgnt_ops.create_account_users(nusers=1)
-            self.s3_clean.update(users)
-            self.test_prefix = 'test-34090'
+            LOGGER.info("Step 2: Shutdown %s data pod by deleting deployment and "
+                        "verify cluster & remaining pods status", pod_name)
+            resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+                master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+                down_method=const.RESTORE_DEPLOYMENT_K8S, delete_pod=pod_name)
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:1512]: pod name will be different after every restart, deployment name will be same.
That needs to be taken care.

Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:1512]: ITs already taken care as last step in this loop:
  pod_name = self.node_master_list[0].get_recent_pod_name(
                deployment_name=self.deployment_name)

**Comment by sonalk0209** [comment]
[pr1755.log](https://github.com/Seagate/cortx-test/files/9120400/pr1755.log)
Collection log

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 2]
```diff
@@ -1344,89 +1343,110 @@ def test_different_data_pods_restart_loop(self):
...
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 7: WRITE/READ/Verify completed successfully.")
+            if CMN_CFG["dtm0_disabled"]:
+                LOGGER.info("Step 8: Create IAM user, create multiple buckets and run IOs")
+                users_rst = self.mgnt_ops.create_account_users(nusers=1)
+                self.s3_clean.update(users_rst)
+                self.test_prefix = 'test-34091-restart'
+                resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users_rst.values())[0],
+                                                            log_prefix=self.test_prefix,
+                                                            skipcleanup=True)
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:1441]: setup_s3bench=False

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 3]
```diff
@@ -1435,93 +1455,112 @@ def test_same_data_pod_restart_loop(self):
...
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 7: WRITE/READ/Verify completed successfully.")
+            if CMN_CFG["dtm0_disabled"]:
+                LOGGER.info("Step 8: Create IAM user, create multiple buckets and run IOs")
+                users_rst = self.mgnt_ops.create_account_users(nusers=1)
+                self.s3_clean.update(users_rst)
+                self.test_prefix = 'test-34090-restart'
+                resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users_rst.values())[0],
+                                                            log_prefix=self.test_prefix,
+                                                            skipcleanup=True)
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:1556]: setup_s3bench=False

**Review by ApurwaMohite — APPROVED** [review]

**Review by ankita-93 — APPROVED** [review]

**Review by rahul-hatwar — APPROVED** [review]
LGTM

**Review by ketanarlulkar — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**